Source/JavaScriptCore/runtime/Structure.cpp

```diff
@@ -672,15 +675,34 @@ Structure* Structure::changePrototypeTransition(VM& vm, Structure* structure, JS
     ASSERT(isValidPrototype(prototype));
 
     DeferGC deferGC(vm);
-    Structure* transition = Structure::create(vm, structure, &deferred);
+    JSObject* key = prototype.isNull() ? nullptr : asObject(prototype);
 
-    transition->m_prototype.set(vm, transition, prototype);
+    bool shouldChain = !structure->hasPolyProto() && structure->typeInfo().type() != GlobalObjectType && !structure->hasBeenDictionary();
+    if (shouldChain) {
+        ASSERT(structure->isObject());
+        if (Structure* existingTransition = structure->m_transitionTable.get(key, 0, TransitionKind::ChangePrototype)) {
+            existingTransition->checkOffsetConsistency();
```

Comment:
Can we add a `ASSERT(!existingTransition->hasPolyProto());` here?

Reply:
Done.